File: VerificarInvariante.java
import java.io.*;
import java.util.regex.*;

public class VerificarInvariante {
    public static void main(String[] args) {
        String logPath = "petri_log.txt";
        boolean invarianteOk = true;
        Pattern pattern = Pattern.compile("P0:(\\d+).*P1:(\\d+).*P2:(\\d+)");
        try (BufferedReader br = new BufferedReader(new FileReader(logPath))) {
            String line;
            while ((line = br.readLine()) != null) {
                Matcher matcher = pattern.matcher(line);
                if (matcher.find()) {
                    int p0 = Integer.parseInt(matcher.group(1));
                    int p1 = Integer.parseInt(matcher.group(2));
                    int p2 = Integer.parseInt(matcher.group(3));
                    if (p0 + p1 + p2 != 3) {
                        System.out.println("Invariante violado en línea: " + line.trim());
                        invarianteOk = false;
                    }
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        if (invarianteOk) {
            System.out.println("¡El invariante de plaza se cumple en todo el log!");
        } else {
            System.out.println("Hubo violaciones del invariante.");
        }
    }
}

File: verificar_invariante.py
import re

def verificar_invariante_plaza(log_path):
    invariante_ok = True
    with open(log_path) as f:
        for line in f:
            match = re.search(r"P0:(\d+).*P1:(\d+).*P2:(\d+)", line)
            if match:
                p0, p1, p2 = map(int, match.groups())
                if p0 + p1 + p2 != 3:
                    print(f"Invariante violado en línea: {line.strip()}")
                    invariante_ok = False
    if invariante_ok:
        print("¡El invariante de plaza se cumple en todo el log!")
    else:
        print("Hubo violaciones del invariante.")

if __name__ == "__main__":
    verificar_invariante_plaza("petri_log.txt")


#python3 verificar_invariante.py